src/main/java/hudson/plugins/gradle/GradleConsoleLogFilter.java

```diff
@@ -0,0 +1,49 @@
+package hudson.plugins.gradle;
+
+import hudson.Extension;
+import hudson.console.ConsoleLogFilter;
+import hudson.model.FreeStyleBuild;
+import hudson.model.Run;
+import hudson.tasks.Builder;
+
+import java.io.IOException;
+import java.io.OutputStream;
+
+@Extension
+public class GradleConsoleLogFilter extends ConsoleLogFilter implements BuildScanPublishedListener {
```

Comment:
IIUC, this applies the filter to all console output of all builds on the Jenkins instance? I would prefer being able to use this as an opt-in, maybe via a build wrapper. There are probably some performance implications for parsing all logs generating on Jenkins.

Reply:
I've reimplemented this as a `BuildWrapper`.